sxm.pl

```diff
@@ -1289,60 +1238,125 @@ package main;
-                if ($data) {
-                    my $response = HTTP::Response->new(HTTP_OK);
-                    $response->header('Content-Type' => 'application/x-mpegURL');
-                    $response->content($data);
-                    $connection->send_response($response);
-                } else {
-                    $connection->send_error(HTTP_INTERNAL_SERVER_ERROR);
+                # Get request with timeout and disconnection handling
+                eval {
+                    local $SIG{ALRM} = sub { die "Request timeout\n" };
+                    alarm(10);  # 10 second timeout for getting request
+                    $request = $connection->get_request;
+                    alarm(0);
+                };
+                if ($@) {
+                    # Timeout or other error getting request
+                    last REQUEST_LOOP;
                 }
-            } elsif ($path =~ /\.aac$/) {
-                my $segment_path = substr($path, 1); # Remove leading slash
-                my $data = $sxm->get_segment($segment_path);
                 
-                if ($data) {
-                    my $response = HTTP::Response->new(HTTP_OK);
-                    $response->header('Content-Type' => 'audio/x-aac');
-                    $response->content($data);
-                    $connection->send_response($response);
-                } else {
-                    $connection->send_error(HTTP_INTERNAL_SERVER_ERROR);
-                }
-            } elsif ($path =~ /\/key\/1$/) {
-                my $response = HTTP::Response->new(HTTP_OK);
-                $response->header('Content-Type' => 'text/plain');
-                $response->content($HLS_AES_KEY);
-                $connection->send_response($response);
-            } else {
-                # Send a simple HTML page with instructions
-                my $response = HTTP::Response->new(HTTP_OK);
-                $response->header('Content-Type' => 'text/html');
-                my $html = <<EOT;
+                # No more requests from this connection
+                last REQUEST_LOOP unless $request;
+                
+                # Process the request with comprehensive error handling
+                eval {
+                    my $path = $request->uri->path;
+                    
+                    # Minimal request logging to improve performance
+                    if ($debug >= 2) {
+                        printf "Request: %s %s\n", $request->method, $path;
+                        STDOUT->flush();
+                    }
+                    
+                    my $start_time = time();
+                    
+                    if ($path =~ /\.m3u8$/) {
+                        my ($channel_name) = $path =~ m|/([^/]+)\.m3u8$|;
+                        printf "Playlist: %s\n", $channel_name if $debug;
+                        STDOUT->flush() if $debug;
+                        
+                        my $data = $sxm->get_playlist($channel_name);
```

Comment:
When getting the playlist, we should use the cached url as long as the channel is not changing.  the cache should only expire if the channel changes, or we have to reauthenticate.